docker_cli/_docker_git.py

```diff
@@ -0,0 +1,77 @@
+import os
+from pathlib import Path
+from textwrap import dedent
+
+from ._docker_mamba import micromamba_docker_lines
+from ._url_reader import URLReader
+
+
+def git_extract_dockerfile(
+    base: str,
+    archive_url: str,
+    url_reader: URLReader,
+    folder_path: os.PathLike[str] = Path("repo"),
```

Comment:
```suggestion
    folder_path: str | os.PathLike[str] = Path("repo"),
```

(this requires `from __future__ import annotations` at the top)